pkgs/development/compilers/swift/wrapper/wrapper.sh

```diff
@@ -240,6 +240,11 @@ fi
 # -target, which also happens to work with Swift.
 if [[ -e $cc_wrapper/nix-support/add-local-cc-cflags-before.sh ]]; then
     source $cc_wrapper/nix-support/add-local-cc-cflags-before.sh
+    if [[ "${extraBefore[-1]}" =~ -march=.+ && "${extraBefore[-2]}" != "-Xcc" ]]; then
+        last="${extraBefore[-1]}"
+        extraBefore[-1]="-Xcc"
+        extraBefore+=($last)
+    fi
```

Comment:
maybe extend the for loop below to handle `-march` anywhere in the array? still fragile if cc-wrapper adds more flags.

```bash
for ((i=0; i < ${#extraBefore[@]}; i++));do
    case "${extraBefore[i]}" in
    -target)
        i=$((i + 1))
        # On Darwin only, need to change 'aarch64' to 'arm64'.
        extraBefore[i]="${extraBefore[i]/aarch64-apple-/arm64-apple-}"
        # On Darwin, Swift requires the triple to be annotated with a version.
        # TODO: Assumes macOS.
        extraBefore[i]="${extraBefore[i]/-apple-darwin/-apple-macosx${MACOSX_DEPLOYMENT_TARGET:-11.0}}"
        ;;
    -march=*)
        [[ i -gt 0 && ${extraBefore[i-1]} == -Xcc ]] && continue
        extraBefore=(
            "${extraBefore[@]:0:i}"
            -Xcc
            "${extraBefore[@]:i:${#extraBefore[@]}}"
        )
        i=$((i + 1))
        ;;
    esac
done
```

Reply:
Ah, that's much nicer. Let me test this and if it works, I'll amend and force-push